code/game/objects/machinery/nuclearbomb.dm

```diff
@@ -242,98 +265,95 @@ obj/machinery/nuclearbomb/proc/nukehack_win(mob/user as mob)
-							GLOB.nukes_set_list |= src
-						else
-							GLOB.nukes_set_list -= src
-					else
-						GLOB.nukes_set_list -= src
-						if(!src.lighthack)
-							src.icon_state = "nuclearbomb1"
-				if (href_list["safety"])
-					src.safety = !( src.safety )
-					if(safety)
-						src.timing = 0
-						stop_processing()
-						GLOB.nukes_set_list -= src
-				if (href_list["anchor"])
-
-					if(removal_stage == 5)
-						src.anchored = FALSE
-						visible_message("<span class='warning'> \The [src] makes a highly unpleasant crunching noise. It looks like the anchoring bolts have been cut.</span>")
-						return
-
-					src.anchored = !( src.anchored )
-					if(src.anchored)
-						visible_message("<span class='warning'> With a steely snap, bolts slide out of [src] and anchor it to the flooring.</span>")
-					else
-						visible_message("<span class='warning'> The anchoring bolts slide back into the depths of [src].</span>")
-
-		for(var/mob/M in viewers(1, src))
-			if ((M.client && M.interactee == src))
-				src.attack_hand(M)
-	else
+	if (!(usr.contents.Find(src) || (in_range(src, usr) && istype(loc, /turf))))
 		usr << browse(null, "window=nuclearbomb")
 		return
-	return
 
+	usr.set_interaction(src)
+
+	if (href_list["auth"])
+		if (auth)
+			auth.loc = loc
+			yes_code = FALSE
+			auth = null
+		else
+			var/obj/item/I = usr.get_active_held_item()
+			if (istype(I, /obj/item/disk/nuclear))
+				if(usr.drop_held_item())
+					I.forceMove(src)
+					auth = I
+	if (auth)
+		if (href_list["type"])
+			if (href_list["type"] == "E")
+				if (code == r_code)
+					yes_code = TRUE
+					code = null
+				else
+					code = "ERROR"
+			else
+				if (href_list["type"] == "R")
+					yes_code = FALSE
+					code = null
+				else
+					code += text("[]", href_list["type"])
+					if (length(code) > 5)
+						code = "ERROR"
+		if (yes_code)
+			if (href_list["time"])
+				var/time = text2num(href_list["time"])
+				timeleft += time
+				timeleft = min(max(round(timeleft), 60), 600)
+			if (href_list["timer"])
+				if (timing == -1.0)
+					return
+				if (safety)
+					to_chat(usr, "<span class='warning'>The safety is still on.</span>")
+					return
+				timing = !timing
+				if(timing)
+					GLOB.nukes_set_list |= src
+					start_processing()
+				if(!lighthack)
+					icon_state = (timing) ? "nuclearbomb2" : "nuclearbomb1"
+			if (href_list["safety"])
+				safety = !safety
+				if(safety)
+					timing = FALSE
+					GLOB.nukes_set_list -= src
+					stop_processing()
+				else
+					GLOB.nukes_set_list |= src
+					start_processing()
+			if (href_list["anchor"])
+				if(removal_stage == 5)
+					anchored = FALSE
+					visible_message("<span class='warning'> \The [src] makes a highly unpleasant crunching noise. It looks like the anchoring bolts have been cut.</span>")
+					return
+
+				anchored = !anchored
+				if(anchored)
+					visible_message("<span class='warning'> With a steely snap, bolts slide out of [src] and anchor it to the flooring.</span>")
```

Comment:
Why the random space at the beginning?

Reply:
copy paste error from elsewhere

Reply:
Can you fix it please?

Reply:
Sure but its all over the code base, ill update what i changed though.

Reply:
Thanks!